nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/reporting/AtlasNiFiFlowLineage.java

```diff
@@ -0,0 +1,714 @@
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor ATLAS_NIFI_URL = new PropertyDescriptor.Builder()
+            .name("atlas-nifi-url")
+            .displayName("NiFi URL for Atlas")
+            .description("NiFi URL is used in Atlas to represent this NiFi cluster (or standalone instance)." +
+                    " It is recommended to use one that can be accessible remotely instead of using 'localhost'.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor ATLAS_DEFAULT_CLUSTER_NAME = new PropertyDescriptor.Builder()
+            .name("atlas-default-cluster-name")
+            .displayName("Atlas Default Cluster Name")
+            .description("Cluster name for Atlas entities reported by this ReportingTask." +
+                    " If not specified, 'atlas.cluster.name' in Atlas Configuration File is used." +
+                    " Cluster name mappings can be configured by user defined properties." +
+                    " See additional detail for detail.")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor ATLAS_CONF_CREATE = new PropertyDescriptor.Builder()
+            .name("atlas-conf-create")
+            .displayName("Create Atlas Configuration File")
+            .description("If enabled, 'atlas-application.properties' file will be created in 'Atlas Configuration Directory'" +
+                    " automatically when this processor starts." +
+                    " Note that the existing configuration file will be overwritten.")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .build();
+
+    static final PropertyDescriptor SSL_CONTEXT_SERVICE = new PropertyDescriptor.Builder()
+            .name("ssl-context-service")
+            .displayName("SSL Context Service")
+            .description("Specifies the SSL Context Service to use for communicating with Atlas and Kafka.")
+            .required(false)
+            .identifiesControllerService(SSLContextService.class)
+            .build();
+
+    static final PropertyDescriptor KAFKA_BOOTSTRAP_SERVERS = new PropertyDescriptor.Builder()
+            .name("kafka-bootstrap-servers")
+            .displayName("Kafka Bootstrap Servers")
+            .description("Kafka Bootstrap Servers to send Atlas hook notification messages based on NiFi provenance events." +
+                    " E.g. 'localhost:9092'" +
+                    " NOTE: Once this reporting task has started, restarting NiFi is required to changed this property" +
```

Comment:
Has a JIRA been filed with Atlas to make this more manageable?

Reply:
I have shared this issue with Atlas project.